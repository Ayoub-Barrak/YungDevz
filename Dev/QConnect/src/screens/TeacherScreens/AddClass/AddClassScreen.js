
import React, { Component } from 'react';
import { View, Text, TextInput, Image, ScrollView , StyleSheet } from 'react-native';
import colors from 'config/colors'
import QcActionButton from 'components/QcActionButton'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {addClass} from 'model/actions/addClass'
//import { addClass } from '../../../model/actions/addClass';


class AddClassScreen extends Component {

  state = {
    className: '',
  }

  addNewClass () {
    let classInfo = {
      name: this.state.className,
      students: []
    };

    this.props.addClass(classInfo);

  }

  render() {
    const { navigate } = this.props.navigation;

    return (

      <View ID="addNewClass" style={{
        alignItems: "center",
      }}>

        <Image source={{ uri: 'https://cdn0.iconfinder.com/data/icons/activities-glyph/2048/2154_-_Sitting_in_class-512.png' }}
          style={{
            width: 200,
            height: 200,
            alignItems: 'center',
            borderRadius: 150 / 2,
            justifyContent: 'center'
          }} />

        <TextInput
          style={{

            backgroundColor: colors.lightGrey,
            borderColor: colors.darkGrey,
            width: 250,
            textAlign: "center",
            alignItems: "center",
            justifyContent: "center"

          }}

          placeholder='Write Class Name Here'

          onChangeText={(classInput) => this.setState({

            className: classInput,

          })}


        />

        <Text>Your Class name is {this.state.className}</Text>

        <QcActionButton
          text="Add Class"
          onPress={() => {
            this.addNewClass();
          }}
        />

        <View>
          <Text>there are {this.props.classrooms.classes.length} classes added so far </Text>
        </View>

      </View>

    );
  }



};


const mapStateToProps = (state) => {
  const { classrooms } = state
  return { classrooms }
};

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addClass,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(AddClassScreen);