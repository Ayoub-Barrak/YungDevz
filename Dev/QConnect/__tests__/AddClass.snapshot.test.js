import React from 'react';
import AddClassScreen from '../src/screens/TeacherScreens/AddClass/AddClassScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<AddClassScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});