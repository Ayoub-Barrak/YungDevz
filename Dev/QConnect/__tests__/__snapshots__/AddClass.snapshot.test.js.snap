// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  ID="addNewClass"
>
  <Text>
    Add a new class placeholder
  </Text>
</View>
`;
